### PR TITLE
Axes doc updates

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Some more docstring updates: 
- ax/hvlines() and axh/vspan()
- boxplot()
- text / annotate methods.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4782,41 +4786,38 @@ def arrow(self, x, y, dx, dy, **kwargs):
         """
         Add an arrow to the axes.
 
-        Draws arrow on specified axis from (`x`, `y`) to (`x` + `dx`,
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: One of these changes is causing:

```
Warning, treated as error:
/home/circleci/project/lib/matplotlib/axes/_axes.py:docstring of matplotlib.axes.Axes.arrow:31:Block quote ends without a blank line; unexpected unindent.
Makefile:20: recipe for target 'html' failed
```

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -3525,6 +3525,10 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
 
           - ``means``: points or lines representing the means.
 
+        Notes
+        -----
+        .. [Notes section required for data comment. See #10189.]
```
Comment (dopplershift) [inline lib/matplotlib/axes/_axes.py]: I'm guessing this is the source of the problem.

Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: No, it was an indented `%(FancyArrow)s`. For now, I've unindented it again, which messes up the Other Parameters section, but at least builds. Might be fixed with #10364.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -4782,41 +4786,38 @@ def arrow(self, x, y, dx, dy, **kwargs):
         """
         Add an arrow to the axes.
 
-        Draws arrow on specified axis from (`x`, `y`) to (`x` + `dx`,
-        `y` + `dy`). Uses FancyArrow patch to construct the arrow.
+        This draws an arrow from (`x`, `y`) to (`x+dx`, `y+dy`).
```
Comment (anntzer) [inline lib/matplotlib/axes/_axes.py]: double quotes outside, or asterisks inside...

**Pull request merged**